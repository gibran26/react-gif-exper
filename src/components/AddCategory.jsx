import { useState } from "react"
import PropTypres from "prop-types";


export const AddCategory = ({ onAddCategory }) => {

    const [inputValue, setInputValue] = useState('')
    
    // Se utiliza para permitir editar el texto
    const onInputChanged = ( {target}) => {
        setInputValue( target.value );
    }

    const onSubmit = (e) => {
        e.preventDefault();
        if( inputValue.trim().length <=1 ) return;

        // setCategories( (categories) =>  [ inputValue, ...categories ]);
        onAddCategory(inputValue.trim());
        setInputValue('');
    }

  return (
    <form onSubmit={onSubmit} aria-label="form">
        <input 
            type="text"
            placeholder="Buscar Gifs"
            value={inputValue}
            onChange={onInputChanged}
        />
    </form>
  )
}

AddCategory.propTypes = {
    onAddCategory: PropTypres.func.isRequired
}
