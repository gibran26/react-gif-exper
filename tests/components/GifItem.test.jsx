import { render } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('GifItem.test pruebas', () => { 
    const titleTest = 'titulo de prueba';
    const urlTest = 'http://mytests.com';

    test('Debe hacer match con el snapshot', () => { 
        const { container } = render(<GifItem title={ titleTest } url={ urlTest }/>);
        expect( container ).toMatchSnapshot();
     });
 })