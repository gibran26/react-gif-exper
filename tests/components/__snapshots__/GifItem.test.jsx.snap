// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem.test pruebas Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      src="http://mytests.com"
    />
    <p>
      titulo de prueba
    </p>
  </div>
</div>
`;
